frontend/src/models/player.js

```diff
@@ -119,6 +128,11 @@ export class Player {
       throw new Error("Invalid announcement");
     }
 
+    // This check is necessary for testing purposes.
+    // As long as game instance isn't set, we can't call affinityEvent
+    if (Object.keys(this.game).length > 0)
+      this.game.affinityEvent(affinityEvents.announcement, this);
```

Comment:
Is there a specific point where the game instance not being set has been a problem? I'd love to avoid test-only logic since that's usually an indicator that we're not testing the real thing. Most often, we actually have a game instance floating around in our tests (even if that makes things a little bloated). Sometimes, it could be a compromise to pass a game stub - in this case a game that's simply stubbing the `affinityEvent()` method so the tests can move along.

If you can point me to a place where this has been a problem, I'm happy to take a look.

Reply:
Affinity Events are called within the game class and we can't test if they're changed without a game class calling the corresponding event. Affinity Events are called when cards are added to the trick or announcements are made. We would need to update every tests where we have such calls.

I don't like this very much either, since:

1. It doesn't feel like a unit test, because so many classes are mandatory it's looking like an integration test.
2. It's really only for tests to not complain...

I can see two possibilities here:
Either refactor now or do it later when affinities are implemented. Can't believe I'm writing this, because we all know what it means 👎🏻

Reply:
I stubbed `game.affinityEvent()` in `party.test.js` which seems to do the trick.